p2p/src/authenticated/discovery/actors/peer/actor.rs

```diff
@@ -159,11 +211,13 @@ impl<E: Spawner + Clock + ReasonablyRealtime + Rng + CryptoRng + Metrics, C: Pub
                     },
                     msg_high = self.high.next() => {
                         let msg = Self::validate_outbound_msg(msg_high, &rate_limits)?;
+                        Self::apply_outbound_rate_limit(&context, &outbound_rate_limits, &self.outbound_rate_limited, &peer, msg.channel).await;
```

Comment:
What I had in mind here was instead some sort of feedback to indicate we should backoff. Sleeping here is pretty suboptimal (as it blocks all messages sent across all channels when we exceed our limit -> whereas when we have access to the governor upstream we can avoid getting in this situation and keep sending on other channels).